packages/datadog-plugin-hapi/src/index.js

```diff
@@ -49,9 +49,42 @@ function createWrapDispatch (tracer, config) {
 }
 
 module.exports = [
+  {
+    name: '@hapi/hapi',
+    versions: ['>=18.2'],
+    file: 'lib/request.js',
+    patch (Request, tracer, config) {
+      this.wrap(Request, 'generate', createWrapGenerate(tracer, config))
+    },
+    unpatch (Request) {
+      this.unwrap(Request, 'generate')
+    }
+  },
+  {
+    name: '@hapi/hapi',
+    versions: ['>=17.9 <18'],
```

Comment:
This block and the previous one could be combined in a single rule with a simpler version range since the `@hapi/hapi` package only contains newer versions.